src/chainstate/stacks/db/transactions.rs

```diff
@@ -874,11 +874,13 @@ impl StacksChainState {
                 // tx fee.
                 let contract_id = contract_call.to_clarity_contract_id();
                 let cost_before = clarity_tx.cost_so_far();
+                let sponsor = tx.sponsor_address_as_optional_data();
 
                 let contract_call_resp = clarity_tx.run_contract_call(
                     &origin_account.principal,
-                    &contract_id,
+                    &sponsor,
                     &contract_call.function_name,
+                    &contract_id,
```

Comment:
Not a bug, just suggested API consistency -- in Clarity and elsewhere in the codebase, the order is `contract_id, function_name, function_args`.  Can you update `run_contract_call()` to reflect this?  Thanks!

Reply:
Also, can you make this the case in `execute_transaction()` below?